cashu/wallet/wallet.py

```diff
@@ -129,8 +148,14 @@ def _construct_proofs(
                 amount=promise.amount,
                 C=C.serialize().hex(),
                 secret=secret,
+                dleq=promise.dleq,
             )
+            if proof.dleq:
+                proof.dleq.C_ = promise.C_
             proofs.append(proof)
+
+        # DLEQ verify
+        self.verify_proofs_dleq(proofs)

```

Comment:
`Alice` verifies dleq when she unblinds a signature (and a dleq proof is present).